app/models/atmosphere/compute_site.rb

```diff
@@ -80,6 +80,8 @@ class ComputeSite < ActiveRecord::Base
               presence: true,
               inclusion: %w(openstack aws rackspace google_compute)
 
+    validate :nic_provider_class_defined, unless: "nic_provider_class_name.nil?"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.